apps/dot_ext/forms.py

```diff
@@ -113,6 +117,13 @@ def clean_agree(self):
         if not agree:
             msg = _('You must agree to the API Terms of Service Agreement')
             raise forms.ValidationError(msg)
+
+        # Log the TOS that the app developer agreed to.
+        app = get_application_model().objects.get(pk=self.instance.pk)
```

Comment:
could you use `self.instance` instead of getting the app from the db again? (i'm not sure what `self.instance` is but I'd expect it to be the Application?)